Install woocommerce-analytics plugin within the promotion banner

ianlin opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

pfKYZu-2ur-p2#comment-2449

This PR tweaks the WC Analytics plugin promotion banner so the plugin can be installed and activated within the WC Admin, rather than open a new tab and send the merchants to WC Analytics product page in WPORG.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Before testing, ensure the banner can be displayed

1. Make sure `woocommerce-analytics` plugin is not installed
2. Set `woocommerce_remote_variant_assignment` to `1` by running either
    - `wp option update woocommerce_remote_variant_assignment 1`, or 
    - `UPDATE wp_options SET option_value = 1 WHERE option_name = 'woocommerce_remote_variant_assignment';`
3. Make sure the banner is not dismissed:
    - ~`DELETE FROM wp_options WHERE option_name = 'woocommerce_order_attribution_install_banner_dismissed';`~
    - **Update**: The commit 3e8e3b5 changed to store the dismissed info in user meta instead of the option, based on the suggestion in the review comment: https://github.com/woocommerce/woocommerce/pull/55715#pullrequestreview-2632338397
    - So please delete the user meta from the `wp_usermeta` table with `meta_key` = `woocommerce_admin_order_attribution_install_banner_dismissed` that associated with your user ID.

#### Test steps

**Overview page - banner**

https://github.com/user-attachments/assets/95673aca-f2df-4b7b-85f5-f6fbfe560614

1. Go to `/wp-admin/admin.php?page=wc-admin&path=/analytics/overview`
2. Click `Try it now`
3. The `Try it now` button become busy and `Try it now` and `Dismiss` buttons are disabled
4. An notice will display when install successfully
5. You will be redirected to `Analytics > Order Attribution` page (`admin.php?page=wc-admin&path=/analytics/order-attribution`)

**Overview page - header**

https://github.com/user-attachments/assets/cf6611ef-4196-4e5c-bf83-7ae436a15d53

1. Remove `woocommerce-analytics` plugin
2. Go to `/wp-admin/admin.php?page=wc-admin&path=/analytics/overview`
3. Click `Dismiss` button
4. The banner should be gone, and there is a `Try Order Attribution` button in the top-right corder
5. Open another incognito page and login as another admin user
6. Go to `/wp-admin/admin.php?page=wc-admin&path=/analytics/overview`, the big banner is still showing
7. Go back to the page that logged in as the previous admin user
8. Click `Try Order Attribution` button in the top-right corder
9. The `Try Order Attribution` button become busy and disabled
10. An notice will display when install successfully
11. You will be redirected to `Analytics > Order Attribution` page (`admin.php?page=wc-admin&path=/analytics/order-attribution`)

**Order editing page**

https://github.com/user-attachments/assets/9266af36-77a8-4619-8deb-4283294e14c9

1. Remove `woocommerce-analytics` plugin
2. Go to `WooCommerce > Orders` and click to enter any order editor, e.g. `/wp-admin/admin.php?page=wc-orders&action=edit&id=284` 
3. Click `Install the extension` button under the Order Attribution metabox in the right hand side
4. The `Install the extension` button become busy and disabled
5. An notice will display when install successfully
6. You will be redirected to `Analytics > Order Attribution` page (`admin.php?page=wc-admin&path=/analytics/order-attribution`)

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Install woocommerce-analytics plugin within the promotion banner

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

ianlin committed:
Install wc-analytics plugin within the promotion banner

ianlin committed:
Remove the external icon on OA promotion banner

ianlin committed:
Merge branch 'trunk' into tweak/install-plugin-within-order-attribution-promotion-banner

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

ianlin committed:
Merge branch 'trunk' into tweak/install-plugin-within-order-attribution-promotion-banner

budzanowski commented:
Looks good. We need clarification on the CTA copy: we should use the same for each scenario.

adrianduffell reviewed (commented):
Nice work @ianlin , hope you don't mind the driveby review.  

I tested the flow from core analytics screen. Overall it's working pretty great 🚀 

A couple of thoughts that could be improved now or in a follow-up targeting the following release:

* Keep the banner and loader animation active during the redirect. 

At the moment I found it confusing how the banner suddenly disappears and the redirection takes a few seconds. See 4s-7s in the video.

https://github.com/user-attachments/assets/cbfb2171-7a19-4c0a-a5e5-2760e02204d7


*  Use user meta to store the dismiss value. 

The banner may be more useful to one admin but not others. We wouldn't want a user performing order fulfilment to dismiss the banner and prevent the main store owner from seeing it, for example. By having it as a user preference we can ensure all admins see it.

Also the proliferation of option values can bloat the store database, and cause it to be autoloaded on front-end requests etc. One time isn't a big performance issue, but if it's done for every WooCommerce banner, it would add up.

ianlin committed:
Keep displaying banner and loading state when redirecting to Order
Attribution analytics page

ianlin commented:
Thanks @adrianduffell, those are really nice suggestions. I added a commit to keep the banner and loader animation active during the redirect. For the user meta I'd need to check how to use that as I have no experience with it. I will see if I can make it next Monday before WC 9.8 code freeze otherwise I will do it as a follow up for the next release.

ianlin committed:
Merge branch 'trunk' into tweak/install-plugin-within-order-attribution-promotion-banner

layoutd reviewed (approved):
Everything looks ok. Thanks for the clear test instructions.

ianlin committed:
Store wc-analytics promotion banner dismissed info in user meta

Instead of storing it in the options.

ianlin commented:
@adrianduffell Thanks again for the nice suggestions. I added two commits that implemented:

- c558196 keep displaying banner and loading state when redirecting to Order Attribution analytics page.
- 3e8e3b5 store wc-analytics promotion banner dismissed info in user meta instead of storing it in the options.

It'd be great if you can take another look, thanks.

ianlin committed:
Lower the fallback percentage for wc-analytics promotion banner

So we won't display to 100% of the merchants when the data from
woocommerce.com is unreachable.

adrianduffell reviewed (approved):
@ianlin Thanks for adding the suggestions, it's working super nice for me now 🚀
